src/Compilers/Core/Portable/Syntax/GreenNode.cs

```diff
@@ -904,16 +904,14 @@ private static bool EquivalentToInternal(GreenNode node1, GreenNode node2)
         public abstract SyntaxToken CreateSeparator<TNode>(SyntaxNode element) where TNode : SyntaxNode;
         public abstract bool IsTriviaWithEndOfLine(); // trivia node has end of line
 
-        public static GreenNode? CreateList(IEnumerable<GreenNode>? nodes, bool alwaysCreateListNode = false)
+        public static GreenNode? CreateList<TFrom>(IReadOnlyList<TFrom>? list, Func<TFrom, GreenNode> select, bool alwaysCreateListNode = false)
```

Comment:
```suggestion
        public static GreenNode? CreateList<TFrom>(IReadOnlyList<TFrom>? list, Func<TFrom, GreenNode> select, bool alwaysCreateListNode = false)
```

curious if we have to take `list` in by interface, and not some concreate type.

Reply:
Not necessarily, it could be List<TFrom> but that would make it much less likely for the efficient branch in the trivia methods to be hit.
And I'm not sure what the advantage would be